File: app/src/main/java/com/examples/android/androidtrainingbzu/MotionEventActivity.java
package com.examples.android.androidtrainingbzu;

import android.content.DialogInterface;
import android.os.Bundle;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.RelativeLayout;

public class MotionEventActivity extends AppCompatActivity
        implements View.OnTouchListener {
    int clickCount;
    long startTime = 0;
    private ViewGroup RootLayout;
    private int Position_X;
    private int Position_Y;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_motion_event);
        RootLayout = (ViewGroup) findViewById(R.id.rootLayout);


        //new image
        Button NewImage = (Button) findViewById(R.id.new_image_button);
        NewImage.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Add_Image();
            }
        });

        clickCount = 0;

    }


    private void Add_Image() {
        final ImageView iv = new ImageView(this);
        iv.setImageResource(R.drawable.bar_chart);

        RelativeLayout.LayoutParams layoutParams = new RelativeLayout.LayoutParams(150, 150);
        iv.setLayoutParams(layoutParams);
        RootLayout.addView(iv, layoutParams);
        iv.setOnTouchListener(this);
    }


    public boolean onTouch(final View view, MotionEvent event) {
        final int X = (int) event.getRawX();
        final int Y = (int) event.getRawY();

        int pointerCount = event.getPointerCount();


        switch (event.getAction() & MotionEvent.ACTION_MASK) {

            case MotionEvent.ACTION_DOWN:
                RelativeLayout.LayoutParams layoutParams = (RelativeLayout.LayoutParams) view.getLayoutParams();
                Position_X = X - layoutParams.leftMargin;
                Position_Y = Y - layoutParams.topMargin;
                break;

            case MotionEvent.ACTION_UP:
                if (startTime == 0) {

                    startTime = System.currentTimeMillis();

                } else {
                    if (System.currentTimeMillis() - startTime < 200) {

                        AlertDialog.Builder builder = new AlertDialog.Builder(MotionEventActivity.this);
                        builder.setMessage("Are you sure you want to delete this?");
                        builder.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialog, int which) {

                                view.setVisibility(View.GONE);

                            }
                        });

                        builder.setNegativeButton("No", new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialog, int which) {
                                dialog.dismiss();
                            }
                        });

                        AlertDialog alertDialog = builder.create();
                        alertDialog.show();

                    }

                    startTime = System.currentTimeMillis();

                }
                break;
            case MotionEvent.ACTION_POINTER_DOWN:
                break;

            case MotionEvent.ACTION_POINTER_UP:
                break;

            case MotionEvent.ACTION_MOVE:

                if (pointerCount == 1) {
                    RelativeLayout.LayoutParams Params = (RelativeLayout.LayoutParams) view.getLayoutParams();
                    Params.leftMargin = X - Position_X;
                    Params.topMargin = Y - Position_Y;
                    Params.rightMargin = -500;
                    Params.bottomMargin = -500;
                    view.setLayoutParams(Params);
                }

                if (pointerCount == 2) {

                    RelativeLayout.LayoutParams layoutParams1 = (RelativeLayout.LayoutParams) view.getLayoutParams();
                    layoutParams1.width = Position_X + (int) event.getX();
                    layoutParams1.height = Position_Y + (int) event.getY();
                    view.setLayoutParams(layoutParams1);
                }

                //Rotation
                if (pointerCount == 3) {
                    //Rotate the ImageView
                    view.setRotation(view.getRotation() + 10.0f);
                }

                break;
        }

// Schedules a repaint for the root Layout.
        RootLayout.invalidate();
        return true;
    }
}

File: app/src/main/java/com/examples/android/androidtrainingbzu/MotionEventActivity.kt
package com.examples.android.androidtrainingbzu

import android.os.Bundle
import android.view.MotionEvent
import android.view.View
import android.view.View.OnTouchListener
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.RelativeLayout
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity

class MotionEventActivity : AppCompatActivity(), OnTouchListener {
    var clickCount = 0
    var startTime: Long = 0
    private var RootLayout: ViewGroup? = null
    private var Position_X = 0
    private var Position_Y = 0
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_motion_event)
        RootLayout = findViewById<View>(R.id.rootLayout) as ViewGroup


        //new image
        val NewImage = findViewById<View>(R.id.new_image_button) as Button
        NewImage.setOnClickListener { Add_Image() }
        clickCount = 0
    }

    private fun Add_Image() {
        val iv = ImageView(this)
        iv.setImageResource(R.drawable.bar_chart)
        val layoutParams = RelativeLayout.LayoutParams(150, 150)
        iv.layoutParams = layoutParams
        RootLayout!!.addView(iv, layoutParams)
        iv.setOnTouchListener(this)
    }

    override fun onTouch(view: View, event: MotionEvent): Boolean {
        val X = event.rawX.toInt()
        val Y = event.rawY.toInt()
        val pointerCount = event.pointerCount
        when (event.action and MotionEvent.ACTION_MASK) {
            MotionEvent.ACTION_DOWN -> {
                val layoutParams = view.layoutParams as RelativeLayout.LayoutParams
                Position_X = X - layoutParams.leftMargin
                Position_Y = Y - layoutParams.topMargin
            }
            MotionEvent.ACTION_UP -> if (startTime == 0L) {
                startTime = System.currentTimeMillis()
            } else {
                if (System.currentTimeMillis() - startTime < 200) {
                    val builder = AlertDialog.Builder(this@MotionEventActivity)
                    builder.setMessage("Are you sure you want to delete this?")
                    builder.setPositiveButton("Yes") { dialog, which ->
                        view.visibility = View.GONE
                    }
                    builder.setNegativeButton("No") { dialog, which -> dialog.dismiss() }
                    val alertDialog = builder.create()
                    alertDialog.show()
                }
                startTime = System.currentTimeMillis()
            }
            MotionEvent.ACTION_POINTER_DOWN -> {}
            MotionEvent.ACTION_POINTER_UP -> {}
            MotionEvent.ACTION_MOVE -> {
                if (pointerCount == 1) {
                    val Params = view.layoutParams as RelativeLayout.LayoutParams
                    Params.leftMargin = X - Position_X
                    Params.topMargin = Y - Position_Y
                    Params.rightMargin = -500
                    Params.bottomMargin = -500
                    view.layoutParams = Params
                }
                if (pointerCount == 2) {
                    val layoutParams1 = view.layoutParams as RelativeLayout.LayoutParams
                    layoutParams1.width = Position_X + event.x.toInt()
                    layoutParams1.height = Position_Y + event.y.toInt()
                    view.layoutParams = layoutParams1
                }

                //Rotation
                if (pointerCount == 3) {
                    //Rotate the ImageView
                    view.rotation = view.rotation + 10.0f
                }
            }
        }

// Schedules a repaint for the root Layout.
        RootLayout!!.invalidate()
        return true
    }
}